packages/stream-metadata/tests/integration/userTips.test.ts

```diff
@@ -0,0 +1,156 @@
+	const baseURL = getTestServerUrl()
+	log('baseURL', baseURL)
+
+	const riverConfig = makeRiverConfig()
+	const bobIdentity = new Bot(undefined, riverConfig)
+	const aliceIdentity = new Bot(undefined, riverConfig)
+	let bob: SyncAgent
+	let alice: SyncAgent
+
+	beforeAll(async () => {
+		await Promise.all([bobIdentity.fundWallet(), aliceIdentity.fundWallet()])
+
+		bob = await bobIdentity.makeSyncAgent()
+		alice = await aliceIdentity.makeSyncAgent()
+		await Promise.all([bob.start(), alice.start()])
+	})
+
+	afterAll(async () => {
+		await Promise.all([bob.stop(), alice.stop()])
+	})
+
+	it('should return correct tip counts after a user sends a tip', async () => {
+		const bobsUserId = bobIdentity.rootWallet.address
+		const alicesUserId = aliceIdentity.rootWallet.address
+		const tipAmount = 1234n // Define a tip amount
+
+		const { spaceId, defaultChannelId: channelId } = await bob.spaces.createSpace(
+			{ spaceName: 'BlastOff_Tip' },
+			bobIdentity.signer,
+		)
+
+		await alice.spaces.joinSpace(spaceId, aliceIdentity.signer)
+		const channel = alice.spaces.getSpace(spaceId).getChannel(channelId)
+		const { eventId } = await channel.sendMessage('hello bob')
+		const messageId = eventId
+
+		const tokenId = await bob.riverConnection.spaceDapp.getTokenIdOfOwner(
+			spaceId,
+			aliceIdentity.rootWallet.address,
+		)
+		expect(tokenId).toBeDefined()
+
+		log(`Bob (${bobsUserId}) tipping Alice (${alicesUserId}) ${tipAmount} ETH`)
+
+		const tx = await bob.riverConnection.spaceDapp.tip(
+			{
+				spaceId,
+				tokenId: tokenId!,
+				currency: ETH_ADDRESS,
+				amount: tipAmount,
+				messageId,
+				channelId,
+				receiver: alicesUserId,
+			},
+			bobIdentity.signer,
+		)
+
+		// Bob sends a tip to Alice
+		const receipt = await tx.wait(2)
+		const tipEvent = bob.riverConnection.spaceDapp.getTipEvent(
+			'0x0000000000000000000000000000000000000000',
+			receipt,
+			alicesUserId,
+		)
+		if (!tipEvent) throw new Error('no tip event found')
+		await bob.riverConnection.client!.addTransaction_Tip(
+			riverConfig.base.chainConfig.chainId,
+			receipt,
+			tipEvent,
+			alicesUserId,
+		)
+
+		log('Tip transaction mined, waiting for stream sync...')
+
+		// Allow some time for river nodes and clients to sync the stream updates
+		await new Promise((resolve) => setTimeout(resolve, 3000)) // Adjust delay if needed
+
+		// Fetch Alice's tips
+		const aliceRoute = `user/${alicesUserId}/tips`
+		log(`Fetching Alice's tips from ${baseURL}/${aliceRoute}`)
+		const aliceResponse = await axios.get<UserTipsResponsePayload>(`${baseURL}/${aliceRoute}`)
+
+		expect(aliceResponse.status).toBe(200)
+		expect(aliceResponse.headers['content-type']).toContain('application/json')
+		expect(aliceResponse.data.tips.received).toEqual(tipAmount.toString())
+		expect(aliceResponse.data.tips.receivedCount).toEqual('1')
+		expect(aliceResponse.data.tips.sent).toEqual('0')
+		expect(aliceResponse.data.tips.sentCount).toEqual('0')
```

Comment:
received is the tip amount received in the ETH_CURRENCY, what if someone tip in a different currency? are we filtering it in the stream view userContent to be only ETH_CURRENCY?

Reply:
yes that's why it's a map :) right now we only tip in eth in the app, but you can tip in anything